documentation/library_guide/dynamic_selection_api/dynamic_load_policy.rst

```diff
@@ -0,0 +1,231 @@
+Dynamic Load Policy
+###################
+
+The dynamic selection API is an experimental feature in the |onedpl_long| 
+(|onedpl_short|) that selects an *execution resource* based on a chosen 
+*selection policy*. There are several policies provided as part 
+of the API. Policies encapsulate the logic and any associated state needed 
+to make a selection. 
+
+The dynamic load policy tracks the number of submissions currently submitted but not yet completed on each 
+resource and selects the resource that has the fewest unfinished submissions. 
+``dynamic_load_policy`` is useful for offloading kernels of varying cost to devices 
+of varying performance. A load-based assignment may achieve a good load balancing 
+by submitting tasks to a resource that completes work faster.
+
+.. code:: cpp
+
+  namespace oneapi::dpl::experimental {
+  
+    template<typename Backend = sycl_backend> 
+    class dynamic_load_policy {
+    public:
+      // useful types
+      using resource_type = typename Backend::resource_type;
+      using wait_type = typename Backend::wait_type;
+      
+      class selection_type {
+      public:
+        dynamic_load_policy<Backend> get_policy() const;
+        resource_type unwrap() const;
+      };
+      
+      // constructors
+      dynamic_load_policy(deferred_initialization_t);
+      dynamic_load_policy();
+      dynamic_load_policy(const std::vector<resource_type>& u);  
+  
+      // deferred initializer
+      void initialize();
+      void initialize(const std::vector<resource_type>& u);
+                      
+      // queries
+      auto get_resources() const;
+      auto get_submission_group();
+      
+      // other implementaton defined functions...
+    };
+  
+  }
+  
+This policy can be used with all of the dynamic selection functions, such as ``select``, ``submit``,
+and ``submit_and_wait``. It also can used with ``policy_traits``.
+
+Example
+-------
+
+The following example demonstrates a simple approach to send work to more than
+one queue concurrently using ``dynamic_load_policy``. The policy selects the
+resource with the fewest number of unfinished submissions.
+
+.. code:: cpp
+
+  #include <oneapi/dpl/dynamic_selection>
+  #include <sycl/sycl.hpp>
+  #include <iostream>
+
+  const std::size_t N = 10000;
+  namespace ex = oneapi::dpl::experimental;
+
+  void f(sycl::handler& h, float *v);

```

Comment:
Consistency about `&` and `*` are also applicable here. But this is not the important comment.

Reply:
done